modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItSqlApiBaseTest.java

```diff
@@ -315,28 +319,50 @@ public void checkExplicitTxRollback() {
     }
 
     /** Check correctness of rw and ro transactions for table scan. */
-    @Test
-    public void checkMixedTransactionsForTable() {
+    @CartesianTest(name = "roTx={0} commit={1} explicit={2}")
+    public void checkMixedTransactionsForTable(
+            @Values(booleans = {true, false}) boolean roTx,
+            @Values(booleans = {true, false}) boolean commit,
+            @Values(booleans = {true, false}) boolean explicit
+    ) {
         sql("CREATE TABLE TEST(ID INT PRIMARY KEY, VAL0 INT)");
 
         Matcher<String> planMatcher = containsTableScan("PUBLIC", "TEST");
 
-        checkMixedTransactions(planMatcher);
+        checkMixedTransactions(planMatcher, roTx, commit, explicit);
     }
 
 
     /** Check correctness of rw and ro transactions for index scan. */
-    @Test
-    public void checkMixedTransactionsForIndex() throws Exception {
+    @CartesianTest(name = "roTx={0} commit={1} explicit={2}")
+    public void checkMixedTransactionsForIndex(
+            @Values(booleans = {true, false}) boolean roTx,
+            @Values(booleans = {true, false}) boolean commit,
+            @Values(booleans = {true, false}) boolean explicit
+    ) throws InterruptedException {
         sql("CREATE TABLE TEST(ID INT PRIMARY KEY, VAL0 INT)");
         sql("CREATE INDEX TEST_IDX ON TEST(VAL0)");
 
+        // Wait for the index to be created.
+        assertTrue(waitForCondition(
```

Comment:
please check `org.apache.ignite.internal.ClusterPerClassIntegrationTest#sql(int, org.apache.ignite.tx.Transaction, java.lang.String, java.lang.Object[])`. It already should awaiting index availability

Reply:
Yes, but in this test we call `assertQuery` to check the plan, which does not use the `sql` method

Reply:
We awaiting index availability after execute create an index. So `assertQuery` doesn't require additional awaiting.

Reply:
Oh, I understand, what you are talking about. Let me investigate, why that check doesn't work